api/v1alpha1/instaslice_types.go

```diff
@@ -44,54 +58,103 @@ const (
-	Memory             int64            `json:"memory"`
+	// +required
+	Allocationstatus AllocationStatus `json:"allocationStatus"`
+	// resourceIdentifier represents uuid used for creating configmap resource
+	// +required
+	Resourceidentifier string `json:"resourceIdentifier"`
+	// namespace represents namespace of user workload
+	// +required
+	Namespace string `json:"namespace"`
+	// podName represents name of the user workload pod
+	// +required
+	PodName string `json:"podName"`
+	// cpu represents amount of cpu requested by user workload
+	// +required
+	Cpu int64 `json:"cpu"`
+	// memory represents amount of memory requested by user workload
+	// +required
+	Memory int64 `json:"memory"`
 }
 
 // InstasliceSpec defines the desired state of Instaslice
 type InstasliceSpec struct {
-	MigGPUUUID         map[string]string            `json:"MigGPUUUID,omitempty"`
-	Allocations        map[string]AllocationDetails `json:"allocations,omitempty"`
-	Migplacement       []Mig                        `json:"migplacement,omitempty"`
-	CpuOnNodeAtBoot    int64                        `json:"cpuonnodeatboot,omitempty"`
-	MemoryOnNodeAtBoot int64                        `json:"memoryonnodeatboot,omitempty"`
+	// migGpuUuid represents uuid of the mig device created on the gpu
+	// +required
+	MigGPUUUID map[string]string `json:"migGpuUuid"`
+	// allocations represents allocation details of user workloads
+	// +required
+	Allocations map[string]AllocationDetails `json:"allocations"`
+	// migplacement represents gpu instance, compute instance with placement for a profile
+	// +required
+	Migplacement []Mig `json:"migplacement"`
+	// cpuonnodeatboot represents total amount of cpu present on the node
+	// +required
+	CpuOnNodeAtBoot int64 `json:"cpuonnodeatboot"`
+	// memoryonnodeatboot represents total amount of memory present on the node
+	// +required
+	MemoryOnNodeAtBoot int64 `json:"memoryonnodeatboot"`
 }
 
 // InstasliceStatus defines the observed state of Instaslice
 type InstasliceStatus struct {
-	Processed bool `json:"processed,omitempty"`
+	// processed represents state of the instaslice object after daemonset creation
+	// +required
+	Processed bool `json:"processed"`
 }
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 
 // Instaslice is the Schema for the instaslices API
 type Instaslice struct {
-	metav1.TypeMeta   `json:",inline"`
+	// TypeMeta contains metadata about the API resource type.
+	// It includes information such as API version and kind.
+	metav1.TypeMeta `json:",inline"`
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   InstasliceSpec   `json:"spec,omitempty"`
-	Status InstasliceStatus `json:"status,omitempty"`
+	// spec defines the current state of different allocations in the Instaslice resource.
+	// It contains configuration details, such as GPU allocation settings,
+	// node-specific parameters, and placement preferences.
+	// +required
+	Spec InstasliceSpec `json:"spec"`
```

Comment:
This doesn't really match with any other operators I see.

Typically spec and status are +optional with omit empty.

Reply:
the operators that we are evaluating do they make allocation decisions? could you please share those operators?

Reply:
FYI, thanks for status suggestion, made it optional

Reply:
Kueue uses omitempty and optional for all its spec/status. 


But I also see deschduler uses required for spec.